src/hotspot/share/ci/ciInlineKlass.cpp

```diff
@@ -78,12 +92,12 @@ bool ciInlineKlass::flatten_array() const {
 
 // Can this inline type be passed as multiple values?
 bool ciInlineKlass::can_be_passed_as_fields() const {
-  GUARDED_VM_ENTRY(return to_InlineKlass()->can_be_passed_as_fields();)
+  GUARDED_VM_ENTRY(return !VectorSupport::skip_value_scalarization(const_cast<ciInlineKlass*>(this)) && to_InlineKlass()->can_be_passed_as_fields();)
 }
 
 // Can this inline type be returned as multiple values?
 bool ciInlineKlass::can_be_returned_as_fields() const {
-  GUARDED_VM_ENTRY(return to_InlineKlass()->can_be_returned_as_fields();)
+  GUARDED_VM_ENTRY(return !VectorSupport::skip_value_scalarization(const_cast<ciInlineKlass*>(this)) && to_InlineKlass()->can_be_returned_as_fields();)
```

Comment:
Why not make vector-specific check a part of `InlineKlass::can_be_returned_as_fields()`?

Reply:
DONE